src/main/java/hudson/plugins/sshslaves/JavaVersionChecker.java

```diff
@@ -88,6 +88,10 @@ private String checkJavaVersion(TaskListener listener, String javaCommand) throw
         StringWriter output = new StringWriter();   // record output from Java
 
         ByteArrayOutputStream out = new ByteArrayOutputStream();
+        // if the command has a space, we need to quote it
+        if(javaCommand.contains(" ")) {
+            javaCommand = "\"" + javaCommand + "\"";
+        }
         connection.exec(javaCommand + " "+ jvmOptions + " -version",out);
```

Comment:
Using Path we can avoid other kinds of issues in the path
you have to add these two imports 

```
java.nio.file.Paths
java.nio.file.Path
```

and apply this patch
```suggestion
        Path javaPath = Paths.get(javaCommand);
        connection.exec(javaPath.toString() + " "+ jvmOptions + " -version",out);
```

Reply:
That does not add quotes around the path if it contains spaces.

Reply:
shit, it is true